test/SILOptimizer/noimplicitcopy_borrowing_parameters.swift

```diff
@@ -0,0 +1,562 @@
+        _ = x
+    }
+    _ = f
+}
+
+func testLoadableBorrowingNonEscapingClosure(_ x: borrowing NonTrivialStruct) {
+    func useNonEscaping(_ f: () -> ()) {}
+    useNonEscaping {
+        _ = x
+    }
+}
+
+func testLoadableBorrowingConsumeOperator(_ x: borrowing NonTrivialStruct) {
+    // expected-error @-1 {{'x' is borrowed and cannot be consumed}}
+    _ = consume x // expected-note {{consumed here}}
+}
+
+func testLoadableBorrowingEnum(_ x: borrowing LoadableEnum) {
+    // expected-error @-1 {{'x' is borrowed and cannot be consumed}}
+    switch x { // expected-note {{consumed here}}
+    case let .x(y):
+        _ = y
+        break
+    case .y:
+        break
+    }
+}
+
+func testLoadableBorrowingCopyOperator(_ x: borrowing NonTrivialStruct) {
+    _ = copy x
+}
+
+//////////////////////////////////////////
+// MARK: Trivial Struct Borrowing Tests //
+//////////////////////////////////////////
+
+func testTrivialBorrowingConsume(_ x: borrowing TrivialStruct) {
+    // expected-error @-1 {{'x' is borrowed and cannot be consumed}}
+    let _ = x // expected-note {{consumed here}}
+    let _ = x.k
+}
+
+func testTrivialBorrowingConsume2(_ x: borrowing TrivialStruct) {
+    // expected-error @-1 {{'x' is borrowed and cannot be consumed}}
+    var y = x // expected-note {{consumed here}}
+    y = x // expected-note {{consumed here}}
+    _ = y
+}
+
+func testTrivialBorrowingConsume3(_ x: borrowing TrivialStruct) {
+    // expected-error @-1 {{'x' is borrowed and cannot be consumed}}
+    let y = x // expected-note {{consumed here}}
+    _ = y
+}
+
+func testTrivialBorrowingUse(_ x: borrowing TrivialStruct) {
+    _ = x
+}
+
+func testTrivialBorrowingUseAndConsume(_ x: borrowing TrivialStruct) {
+    // expected-error @-1 {{'x' is borrowed and cannot be consumed}}
+    _ = x
+    let _ = x // expected-note {{consumed here}}
+}
+
+func testTrivialBorrowingCallBorrowValDefault(_ x: borrowing TrivialStruct) {
+    borrowValDefault(x)
+}
+
+func testTrivialBorrowingCallBorrowValBorrowing(_ x: borrowing TrivialStruct) {
+    borrowValBorrowing(x)
+}
+
+func testTrivialBorrowingCallMethodSelfDefault(_ x: borrowing TrivialStruct) {
+    x.doSomethingDefault()
+}
+
+func testTrivialBorrowingCallMethodSelfBorrowing(_ x: borrowing TrivialStruct) {
+    x.doSomethingBorrowing()
+}
+
+func testTrivialBorrowingEscapingClosure(_ x: borrowing TrivialStruct) {
+    // expected-error @-1 {{'x' cannot be captured by an escaping closure since it is a borrowed parameter}}
+    var f: () -> () = {}
+    f = { // expected-note {{closure capturing 'x' here}}
+        _ = x
+    }
+    _ = f
+}
+
+func testTrivialBorrowingNonEscapingClosure(_ x: borrowing TrivialStruct) {
+    // expected-error @-1 {{'x' cannot be captured by an escaping closure since it is a borrowed parameter}}
+    // TODO: Wrong
+    func useNonEscaping(_ f: () -> ()) {}
+    useNonEscaping { // expected-note {{closure capturing 'x' here}}
+        _ = x
+    }
+}
```

Comment:
Yes. I am going to fix that in a subsequent commit. This is providing the initial support. There will not be a source break when I fix it since we are removing an error.